src/CMakeLists.txt

```diff
@@ -1,326 +1,131 @@
-  string (REPLACE ";" " " PKG_CONFIG_LIBS_PRIVATE "${PKG_CONFIG_LIBS_PRIVATE}")
-  if (${PLATFORM} MATCHES "Desktop")
-    target_link_libraries(raylib_static PRIVATE glfw ${GLFW_LIBRARIES} ${LIBS_PRIVATE})
-  endif()
-
-  if (WITH_PIC)
-    set_property(TARGET raylib_static PROPERTY POSITION_INDEPENDENT_CODE ON)
-  endif()
-  set_target_properties(raylib_static PROPERTIES PUBLIC_HEADER "raylib.h")
-  if(NOT MSVC) # Keep lib*.(a|dll) name, but avoid *.lib files overwriting each other on Windows
-    set_target_properties(raylib_static PROPERTIES OUTPUT_NAME raylib)
-  endif()
-  install(
-      TARGETS raylib_static
-      ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-      PUBLIC_HEADER DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
-  )
-  set_target_properties(raylib_static PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${CMAKE_CURRENT_BINARY_DIR}")
-
-  add_test("pkg-config--static" ${PROJECT_SOURCE_DIR}/../cmake/test-pkgconfig.sh --static)
-endif(STATIC)
-
-
-if(SHARED)
-  MESSAGE(STATUS "Building raylib shared library")
-  add_library(raylib SHARED ${sources})
-
-  target_compile_definitions(raylib
-      PUBLIC ${PLATFORM_CPP}
-      PUBLIC ${GRAPHICS}
-  )
-
-  if(MSVC)
-    target_compile_definitions(raylib
-      PRIVATE $<BUILD_INTERFACE:BUILD_LIBTYPE_SHARED>
-      INTERFACE $<INSTALL_INTERFACE:USE_LIBTYPE_SHARED>
+# Get the sources together
+set(raylib_public_headers
+    raylib.h
+    rlgl.h
+    physac.h
+    raymath.h
+    raudio.h
     )
-  endif()
-
-  set(PKG_CONFIG_LIBS_EXTRA "")
-
-  set_property(TARGET raylib PROPERTY POSITION_INDEPENDENT_CODE ON)
-  set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_LIBDIR}")
-  set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
-  set(CMAKE_MACOSX_RPATH ON)
-
-  target_link_libraries(raylib ${LIBS_PRIVATE})
-  if (${PLATFORM} MATCHES "Desktop")
-    target_link_libraries(raylib glfw)
-  endif()
-  set_target_properties(raylib PROPERTIES
-      PUBLIC_HEADER "raylib.h"
-      VERSION   ${PROJECT_VERSION}
-      SOVERSION ${API_VERSION}
-  )
 
-  if (WIN32)
-    install(
-      TARGETS raylib
-      ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-      LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-      RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
-      PUBLIC_HEADER DESTINATION "include"
+set(raylib_sources
+    core.c
+    models.c
+    shapes.c
+    text.c
+    textures.c
+    utils.c
     )
-  else()
-    install(
-        TARGETS raylib
-        LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-        PUBLIC_HEADER DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
-    )
-  endif()
-  set_target_properties(raylib PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${CMAKE_CURRENT_BINARY_DIR}")
-
-  add_test("pkg-config" ${PROJECT_SOURCE_DIR}/../cmake/test-pkgconfig.sh)
-else(SHARED)
-  add_library(raylib ALIAS raylib_static)
-endif(SHARED)
 
-if (NOT DEFINED PKG_CONFIG_LIBS_EXTRA)
-  set(PKG_CONFIG_LIBS_EXTRA "${PKG_CONFIG_LIBS_PRIVATE}")
-endif()
+include(GlfwImport)
```

Comment:
Would `find_package` work for this? Use it as a dependency rather than an include?

Reply:
@RobLoach GlfwImport is just a new file taht I copied the logic that included the GLFW library. For segmentation purpose only because GLFW logic contained a lot of if... else conditions and polutes this file. Here you only care to know that the GlfwImport.cmake file is taking care of that - makes it cleaner and easier to reason with.